### DISCUSSION TITLE
Views - filter data for current user

### DISCUSSION THREAD

**Discussion by erakastrati** [Budibase/budibase#2365] [Help] [answered]
Hi there,

1. Is there any possibility to filter data for current user?|
2. Is there any way to show current logged in user on Screens?

Thanks,
Era.

**Comment by kevmodrome** [comment, marked as answer]
Hi Era,

Yes, select the `Current User` option in the data provider.

**Reply by dchuk** [reply]
Hi,

I can't find this option at all in a data provider. Can you provide a screenshot of what you mean?

**Reply by joebudi** [reply]
It's under relationships.

**Comment by ajarosz** [comment]
Let me follow up on this question.

I have following Data design:
Table `Students` with relationship column to 'Lessons' - each student may have multiple lesson records
Table `Lessons` with relationship column to 'Students' - each lesson have one student and one user/teacher assigned
Built-in table `Users` with relationship column to 'Lessons' - each user/teacher may have multiple lesson records

I'd like to have a screen where logged in user/teacher will be presented with all the lessons he had.

So in Data provider I set up `Current User.Lessons` as suggested and the result is that in all returned `Lessons` rows column 'Student' shows up empty (see ss below).  Setting data source to just `Lessons` results in showing all the lessons with correct values for 'Student' column but of course it is not filtered by user. 
I tried setting up filter on `Lessons` table but apparently it is not possible because you can provide only literals there (or is there a way to provide current user as a criteria?) and you cannot filter relationship columns (at least it is not available in filter dropdown).

<img width="1105" alt="2021-08-26_10-00-08" src="https://user-images.githubusercontent.com/5153439/130924759-4a66dc52-a178-4e68-99ed-ef4945fa965c.png">

@kevmodrome is this a known limitation or am I doing something wrong here?

Thanks,
Artur

**Reply by mjashanks** [reply]
Hi Artur. This is a bug. I've logged it here, so that you can track it: https://github.com/Budibase/budibase/issues/2479

Thank you for reporting it - it's an important one.

**Reply by ajarosz** [reply]
Thanks! Sure I will follow on that.
